### PR TITLE
Add integration test script

### PR DESCRIPTION
**Pull request by wajihyassine**
### Description of the change

Integration testing script between dfTimewolf, Timesketch, and Turbinia.

It will run the dfTimewolf gcp_turbinia_ts recipe against the OSDFIR Infra helm chart then check for any failed Plaso tasks from Turbinia or any failed Timesketch timelines imports.

### CODE REVIEW

**Comment by wajihyassine** [comment]
@aarontp @berggren pinging as this has been stale if you can PTAL

File: tools/run-integration-test.sh [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+
+# Forward k8s services
+echo "Forwarding k8s $RELEASE services"
+kubectl --namespace default port-forward service/$RELEASE-timesketch 5000:5000 > /dev/null 2>&1 &
+kubectl --namespace default port-forward service/$RELEASE-turbinia 8000:8000  > /dev/null 2>&1 &
+
+# Back up existing Timesketch configs else script will attempt to connect to wrong Timesketch instance
+if  [ -f ~/.timesketchrc ] && [ -f ~/.timesketch.token ] 
+then
+    echo "Backing up existing Timesketch configs to ~/.timesketchrc.$DATE and ~/.timesketch.token.$DATE"
```
Comment (berggren) [inline tools/run-integration-test.sh:39]: Do we need to keep these?

File: tools/run-integration-test.sh [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+    mv ~/.timesketchrc ~/.timesketchrc.$DATE
+    mv ~/.timesketch.token ~/.timesketch.token.$DATE
+fi
+
+echo "Generating Timesketch config..."
+TS_SECRET=$(kubectl get secret --namespace default $RELEASE-timesketch-secret -o jsonpath="{.data.timesketch-user}" | base64 -d)
+if ! [ -f ~/.timesketchrc ] && ! [ -f ~/.timesketch.token ]; then
+    expect <<END_EXPECT
```
Comment (berggren) [inline tools/run-integration-test.sh:47]: ugh, `expect` .. :) I'll make an issue on the TS CLI side to take these are arguments as well

File: tools/run-integration-test.sh [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+		"API_SERVER_ADDRESS": "http://127.0.0.1",
+		"API_SERVER_PORT": 8000,
+		"API_AUTHENTICATION_ENABLED": false,
+		"CLIENT_SECRETS_FILENAME": ".client_secrets.json",
+		"CREDENTIALS_FILENAME": ".credentials_default.json"
+	}
+}
+EOL
+
+# Run dfTimewolf recipe
```
Comment (berggren) [inline tools/run-integration-test.sh:83]: In the longer run, let's make it easy to run multiple different recipes

File: tools/run-integration-test.sh [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+	plaso_status=$(echo $status | jq '[.tasks[]] | map({name: .name, id: .id, successful: .successful, worker_name: .worker_name}) | map(select(.name == "PlasoParserTask" or .name == "PlasoHasherTask")) | map(select(.successful==false))')
+	length=$(echo $plaso_status | jq '. | length')
+	if [[ $length > 0 ]]
+	then
+	  echo "A failed Plaso Task for Turbinia request $req has been detected."
+	  echo "Listing failed Tasks..."
+    # Grab the Task ID
+    tasks=$(echo $plaso_status | jq -r '.[] | .id')
+	  FAILED=1
+	  for t in $tasks
```
Comment (berggren) [inline tools/run-integration-test.sh:109]: I'm not a big fan of one letter variable names. I opt to call this `task` instead

**Review by berggren — APPROVED** [review]
I have some minor comments and suggestions, but overall LGTM

**Comment by berggren** [comment]
> @aarontp @berggren pinging as this has been stale if you can PTAL

Reviewed. Some smaller comments but nothing blocking. LGTM from my side.

File: tools/run-integration-test.sh (hunk 1 above)
Comment (wajihyassine) [reply tools/run-integration-test.sh:39]: Ehh not really given only should be run within Jenkins anyways. This is more for admin testings/fixes and not having to manually remove the configs each time but I can consider just removing it overall as tests stabilize

File: tools/run-integration-test.sh (hunk 2 above)
Comment (wajihyassine) [reply tools/run-integration-test.sh:47]: Awesome thank you felt a bit hacky using `expect` but works in the meantime..

File: tools/run-integration-test.sh (hunk 3 above)
Comment (wajihyassine) [reply tools/run-integration-test.sh:83]: Yeah agreed, once a solution comes to mind on how to best do that ill update in another pass so we can re-use for other tests :)

File: tools/run-integration-test.sh (hunk 4 above)
Comment (wajihyassine) [reply tools/run-integration-test.sh]: Fair enough, I updated any one character variable (including this) to be more explicit

**Comment by wajihyassine** [comment]
So since you already approved the Turbinia e2e script (in https://github.com/google/turbinia/pull/1400) @aarontp  and this script is the same exact minus the TS part which has been reviewed by Johan I'm going to opt to merge this in so I can close this out from my end before end of the year. Happy to add any feedback/improvements in another PR if necessary

**Pull request merged**